### ISSUE TITLE
Redundant axioms and unnatural type declarations

### ISSUE THREAD

**Issue by SkorikGG** [ekmett/constraints#56]
Axioms `plusZero`, `timesZero`, `timesOne`, `powZero`, `powOne` are redundant, because GHC already reduces these types. For example, in ghci:
```
> :t plusZero
plusZero :: Dict n ~ n
```

In the type declarations for `plusAssociates`, `timesAssociates`, `minAssociates`, `maxAssociates`  the order of appearance of type variables is changed. For this reason, there are difficulties with using  `TypeApplications`. For example,
```
plusAssociates :: forall n m o. Dict (((m + n) + o) ~ (m + (n + o))) 
plusAssociates @m1 @m2 @m3 :: Dict ( ((m2+m1)+m3) ~ (m2+(m1+m3)) )
```

**Comment by RyanGlScott** [comment]
For the first point, I suppose they are redundant. I don't know if we necessarily want to remove them though, since they are useful observations one can make in a library of primitive facts about natural numbers. Perhaps we should change the implementation to `plusZero = Dict`, though, since GHC can deduce this without the use of `axiom`?

For the second point, that seems quite bad. Care to open a PR that puts the type variables in a more sensible order?

**Comment by SkorikGG** [comment]
Pull request [#58](https://github.com/ekmett/constraints/pull/58) is added.

**Comment by RyanGlScott** [comment]
Fixed in 786b9d3bad51d9d967b5583da3aefb42c7ba4ed5.
